core/src/main/scala/kafka/controller/ControllerRequestMerger.scala

```diff
@@ -115,10 +119,22 @@ class ControllerRequestMerger(config: KafkaConfig) extends Logging {
     aggregateTopicIds.putAll(topicIds)
   }
 
+  /**
+   *@return This method should return true if the merging should continue, and false otherwise.
+   */
   private def addLeaderAndIsrRequest(request: LeaderAndIsrRequest.Builder,
-                                     callback: AbstractResponse => Unit): Unit = {
+                                     callback: AbstractResponse => Unit): Boolean = {
     mergeTopicIds(request.topicIds())
 
+    if (request.`type`().equals(LeaderAndIsrRequestType.FULL)) {
+      // There should be no other LeaderAndISR partitions in the merger when a full LeaderAndISR request is sent
+      if (!leaderAndIsrPartitionStates.isEmpty) {
+        throw new IllegalStateException("A Full LeaderAndISR request is received while some LeaderAndISR partitions are already " +
+        "ControllerRequest merger")
```

Comment:
"already ControllerRequest merger"?  Is there a word missing in there?

Reply:
Thanks for catching this. I've fixed the message.